pallets/nfts/src/features/create_delete_item.rs

```diff
@@ -260,6 +264,15 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 		PendingSwapOf::<T, I>::remove(collection, item);
 		ItemAttributesApprovalsOf::<T, I>::remove(collection, item);
 
+		let balance = AccountBalance::<T, I>::get(collection, &owner)
```

Comment:
Perhaps using .take() is better, as you then only need to insert it if balance > zero?